File: frontend/src/app/models/user-profile-details-dto.ts
export interface UserProfileDetailsDto {
  userUuid: string;
  firstname: string;
  surname: string;
  phoneNumber: string;
  username: string;
  password?: string; // Hasło jest opcjonalne lub powinno być pominięte w DTO wysyłanym do klienta
  createdDate: string; // LocalDateTime na backendzie często mapuje się na string w formacie ISO 8601 na froncie
  email: string;
  isActive: boolean;
  name: string; // Nazwa roli
}

File: backend/src/main/java/pl/lodz/p/backend/appointment/dto/UserProfileDetailsDto.java
package pl.lodz.p.backend.appointment.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UserProfileDetailsDto {
    private UUID userUuid;
    private String firstname;
    private String surname;
    private String phoneNumber;
    private String username; // Nowe pole
    private String password;
    private LocalDateTime createdDate;
    private String email;
    private Boolean isActive;
    private String name;
}
